### DISCUSSION TITLE
access data in useState across dynamic routes in React

### DISCUSSION THREAD

**Discussion by ufelgen** [neuefische/questions#394] [Web] [answered]
Hi everyone,

I am (for now) storing my array data in a useState, so that I can update it through a form. I need to access this data in dynamic pages ( [slug].js ) but I cannot seem to find a way to do this. Since my dynamic pages aren't components that are called somewhere, I cannot find a way to pass the data as a prop.

I am grateful for any suggestions. Thanks!

**Comment by ThomasKirst** [comment, marked as answer]
There are basically two *simple* options:
1. You can use a localStorage hook which allows you to store your state into the browser´s local storage. On your dynamic page you can afterwards access that state again by reading from the local storage value. In your capstone-project template there should be a [ready-to-use hook](https://github.com/julialoeschel/capstone-template/blob/main/helpers/hooks.js).

2. Another option is to uplift your state into the `_app.js` file from where you can make your state available to all other pages which need to access it.

As an example:
```js
import { useState } from "react";
import type { AppProps } from "next/app";

function MyApp({ Component, pageProps }: AppProps) {
  const [count, setCount] = useState(0);

  return (
    <Component
      {...pageProps}
      count={count} // This will be passed to all pages
      onIncrement={() => setCount(count + 1)}  // This will be passed to all pages
    />
  );
}

export default MyApp;
```

You can then access these props in any page:

```js
export default function Test({ count }) {
  console.log(count);
  return <h1>Hello</h1>;
}
```

**Reply by ufelgen** [reply]
Thank you!!!!! I tried option number 2 and it worked beautifully. I will try and implement the local storage option tomorrow since it makes more sense for overall functionality.

just a remark: AppProps seems to be typescript-specific. in my setup it worked fine without it.

Thanks again!! 🙏
